### PR TITLE
feat: add watsonx bearer token auth and display model cost

### PR DESCRIPTION
**Pull request by gprem09**

### CODE REVIEW

File: examples/watsonx/README.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
+To get started, set your `WATSONX_AI_APIKEY` or `WATSONX_AI_BEARER_TOKEN`, and `WATSONX_PROJECT_ID` environment variables.
```
Comment (mldangelo) [inline examples/watsonx/README.md]: nit, consider telling people where / how to find these

File: site/docs/providers/watsonx.md [hunk 2]
```diff
@@ -61,7 +61,10 @@ To install the WatsonX provider, use the following steps:
 2. Set up the necessary environment variables:
 
    ```sh
-   export WATSONX_API_KEY=your-ibm-cloud-api-key
+   export WATSONX_AI_APIKEY=your-ibm-cloud-api-key
+   OR
+   export WATSONX_AI_BEARER_TOKEN=your-ibm-cloud-bearer-token
+
    export WATSONX_PROJECT_ID=your-ibm-project-id
    ```
```
Comment (mldangelo) [inline site/docs/providers/watsonx.md]: nit have two blocks that someone can copy paste

File: src/envars.ts [hunk 3]
```diff
@@ -107,8 +107,9 @@ export type EnvVars = {
   REPLICATE_TOP_P?: number;
   VOYAGE_API_BASE_URL?: string;
   VOYAGE_API_KEY?: string;
-  WATSONX_API_KEY?: string;
+  WATSONX_AI_APIKEY?: string;
```
Comment (mldangelo) [inline src/envars.ts:110]: Are you sure you want to rename this?  We have WATSONX_ and WATSONX_AI_ as prefixes? Should we pick one or is there reason to have both?

**Review by mldangelo — APPROVED** [review]
Looks good, I left a few comments. Happy to merge as is if you are happy

File: src/envars.ts (hunk 3 above)
Comment (gprem09) [reply src/envars.ts:110]: We are using WATSONX_AI_. No longer using WATSONX_. I have updated the following changes to it.

**Pull request merged**